pkg/scheduler/scheduler_test.go

```diff
@@ -1894,3 +1904,81 @@ func TestRequeueAndUpdate(t *testing.T) {
 		})
 	}
 }
+
+func TestResourcesToReserve(t *testing.T) {
```

Comment:
I prefer we don't test at this level.

Better incorporate one or 2 test cases in `TestSchedule`.

Reply:
TestSchedule test quite complex logic. I've added integration tests to test the full functionality